### PR TITLE
Remove optimisation to reuse semanticdb jar from the classpath

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would only download mtags module and reuse the semanticDB jars on the classpath, but that can cause other libraries not be fetched and cause unpredictable or easy to miss runtime issues.

Now, we fetch all the transitive dependencies of mtags.

Fixes https://github.com/scalameta/metals/issues/1647

### CODE REVIEW

File: tests/slow/src/test/scala/tests/feature/FoldingCrossLspSuite.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package tests.feature
+
+import tests.BaseLspSuite
+import scala.meta.internal.metals.{BuildInfo => V}
+
+class FoldingCrossLspSuite extends BaseLspSuite("foldingRange-cross") {
```
Comment (kpbochenek) [inline tests/slow/src/test/scala/tests/feature/FoldingCrossLspSuite.scala:6]: Question: Is this test related to mentioned optimisation removal?

Comment (tgodzik) [reply tests/slow/src/test/scala/tests/feature/FoldingCrossLspSuite.scala:6]: Yes actually, this issue came up when using Folding on Scala versions different than `2.12.11`
